File: problems/1657/solution.py
class Solution:
    def closeStrings(self, word1: str, word2: str) -> bool:
        if len(word1) != len(word2):
            return False
        
        cnt1, cnt2 = [0] * 26, [0] * 26
        
        for ch in word1:
            cnt1[ord(ch) - ord('a')] += 1
        
        for ch in word2:
            cnt2[ord(ch) - ord('a')] += 1
            
        for i in range(26):
            if (cnt1[i] > 0 and cnt2[i] == 0) or (cnt1[i] == 0 and cnt2[i] > 0):
                return False
        
        cnt1.sort()
        cnt2.sort()
        
        for i in range(26):
            if cnt1[i] != cnt2[i]:
                return False
        
        return True

File: problems/1657/solution.go
package main
func closeStrings(word1 string, word2 string) bool {
    if len(word1) != len(word2) {
        return false
    }
    
    cnt1, cnt2 := make([]int, 26), make([]int, 26)
    
    for _, ch := range word1 {
        cnt1[int(ch) - int('a')]++
    }
    
    for _, ch := range word2 {
        cnt2[int(ch) - int('a')]++
    }
    
    for i := 0; i < 26; i++ {
        if ((cnt1[i] > 0 && cnt2[i] == 0) || (cnt1[i] == 0 && cnt2[i] > 0)) {
            return false;
        }
    }
    
    sort.Ints(cnt1)
    sort.Ints(cnt2)
    
    for i := 0; i < 26; i++ {
        if (cnt1[i] != cnt2[i]) {
            return false
        }
    }
    
    return true
}